File: src/main/java/it/contrader/service/AbstractService.java
package it.contrader.service;

import java.util.ArrayList;

import java.util.List;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import it.contrader.converter.Converter;
import it.contrader.dao.MyRepository;

@Service
public abstract class AbstractService<Entity,DTO> implements ServiceDTO<Entity,DTO> {
	
	@Autowired
	protected MyRepository<Entity,Long> myRepository;
	@Autowired
	protected Converter<Entity,DTO> converter;
	
	@Override
	public Entity insert(DTO dto) {
		return myRepository.save(converter.toEntity(dto));
	}

	@Override
	public List<DTO> getAll() {
		Iterable<Entity> iter = myRepository.findAll();
		List<DTO> list = new ArrayList<>();
		for (Entity item : iter) {
	        list.add(converter.toDTO(item));
	    }
		return list;
	}
	
	//GETS all entities depending on id of a parameter, see (Entity)Repository
	@Override
	public List<DTO> getAllBy(long id) {
		List<Entity> iter = myRepository.findAllBy(id);
		List<DTO> list = new ArrayList<>();
		for (Entity item : iter) {
	        list.add(converter.toDTO(item));
	    }
		return list;
	}

	@Override
	public DTO read(long id) {
		return converter.toDTO(myRepository.findById(id).get());
	}

	@Override
	public Entity update(DTO dto) {
		return myRepository.save(converter.toEntity(dto));
	}

	@Override
	public void delete(long id) {
		myRepository.deleteById(id);
	}
}

File: src/app/services/abstract.service.ts
import { Service } from './service.service';
import { Observable } from 'rxjs';
import { HttpClient } from '@angular/common/http';

export abstract class AbstractService<DTO> implements Service<DTO> {

    type: string;

    constructor(protected http: HttpClient) {

    }

    getAllBy(id: number): Observable<DTO[]> {
        return this.http.get<DTO[]>('http://localhost:8080/' + this.type + '/getallby?id=' + id);
    }
    getAll(): Observable<DTO[]> {
        return this.http.get<DTO[]>('http://localhost:8080/' + this.type + '/getall');
    }

    read(id: number): Observable<DTO> {
        return this.http.get<DTO>('http://localhost:8080/' + this.type + '/read?id=' + id);
    }

    delete(id: number): Observable<any> {
        return this.http.delete('http://localhost:8080/' + this.type + '/delete?id=' + id);
    }

    insert(dto: DTO): Observable<any> {
        return this.http.post('http://localhost:8080/' + this.type + '/insert', dto);
    }

    update(dto: DTO): Observable<any> {
        return this.http.put('http://localhost:8080/' + this.type + '/update', dto);
    }
}